backend/cmd/headlamp_test.go

```diff
@@ -115,6 +117,27 @@ func getResponse(handler http.Handler, method, url string, body interface{}) (*h
 	return rr, nil
 }
 
+func getResponseFromRestrictedEndpoint(
+	handler http.Handler, method, url string, body interface{},
+	token string,
+) (*httptest.ResponseRecorder, error) {
+	os.Setenv("HEADLAMP_BACKEND_TOKEN", token)
+	defer os.Unsetenv("HEADLAMP_BACKEND_TOKEN")
+
+	req, err := makeJSONReq(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("X-HEADLAMP_BACKEND-TOKEN", token)
+
+	rr := httptest.NewRecorder()
+
+	handler.ServeHTTP(rr, req)
+
+	return rr, nil
+}
+
 //nolint:gocognit,funlen
 func TestDynamicClusters(t *testing.T) {
```

Comment:
Since this is an integration test please make it run only when `HEADLAMP_RUN_INTEGRATION_TESTS` is set to `true`.
Refer:
https://github.com/headlamp-k8s/headlamp/blob/e96003b269153545ff756e74401c44b9587c8fc3/backend/pkg/helm/release_test.go#L37

Reply:
done